### PR TITLE
Upgrade to Unity Analytics

### PR DESCRIPTION
**Pull request by LucaHaverty**
### Description
Switches our current analytics system to unity analytics. Unity analytics is installed via the package manager and can be monitored at [https://dashboard.unity3d.com/](https://dashboard.unity3d.com/). Our analytics manager will also be getting an overhaul as much of it is several years old. The unity analytics manager automatically reports some things such as when people open or close synthesis, but we can also make custom events.

### Objectives
- [x] Install the unity analytics package
- [x] Set up unity dashboard
- [x] Get sending standard events to work
- [x] Get sending custom events to work
- [x] Update all of the old events to unity analytics
- [x] Discuss with Hunter what new events we want to report and implement them
- [x] Register events with params on unity dashboard

### Testing Done
- Tested sending custom and standard events to the unity dashboard
- Tested all new custom events
- Tested sending events in a build

### Current Custom Events
ModalCreated,
        PanelCreated,
        ActiveModalClosed,
        PanelClosed,
        ExitedToMenu,
        SettingsSaved,
        SettingsReset,
        RobotSpawned,
        FieldSpawned,
        MatchStarted,
        MatchEnded,        
DrivetrainSwitched,
ScoringZoneUpdated

If you think of other events to add, message me on discord or comment on this PR

[JIRA Issue](https://jira.autodesk.com/browse/AARD-1496)

### CODE REVIEW

**Review by Markytt — COMMENTED** [review]
When I tried to click the unity link, it brought me to a 404 page.

**Comment by PepperLola** [comment]
Fixed the unity link in the pr description

**Comment by LucaHaverty** [comment]
That's weird. The link works for me even on a browser where I'm not signed in. Try searching unity dashboard and see if it brings you to a different page?

**Comment by PepperLola** [comment]
The markdown had it linking to the URL "url" but now it actually points to dashboard.unity3d.com

**Comment by BrandonPacewic** [comment]
Do we feel comfortable outright deleting the old analytics code?

**Review by BrandonPacewic — COMMENTED** [review]
Do we feel comfortable outright deleting the old analytics code?

**Comment by LucaHaverty** [comment]
Hunter said I could delete it because we didn't really use it much anyway

**Comment by LucaHaverty** [comment]
Oh sorry I misunderstood what you were saying. Yeah I will delete them

File: engine/Assets/Scripts/Analytics/AnalyticsManager.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Synthesis.PreferenceManager;
+using Unity.Services.Analytics;
+using Unity.Services.Core;
+using UnityEngine;
+
+namespace Analytics {
+    // TODO: after testing, disable when in editor
```
Comment (BrandonPacewic) [inline engine/Assets/Scripts/Analytics/AnalyticsManager.cs:12]: Still a todo?

File: engine/Assets/Scripts/Analytics/AnalyticsManager.cs [hunk 2]
```diff
@@ -0,0 +1,145 @@
+// #define DEBUG_ANALYTICS // Uncomment this line to print analytic debug information
```
Comment (BrandonPacewic) [inline engine/Assets/Scripts/Analytics/AnalyticsManager.cs:1]: We leaving this here?

File: engine/Assets/Scripts/Analytics/AnalyticsManager.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            });
+
+            if (!Application.isEditor)
+                await UnityServices.InitializeAsync();
+
+#if DEBUG_ANALYTICS
+            AnalyticsDebug("Unity services initialized");
+#endif
+
+            // TODO: check if the user gives consent to collect information
```
Comment (BrandonPacewic) [inline engine/Assets/Scripts/Analytics/AnalyticsManager.cs:48]: Should we make this part of Hunter tracking down multiplayer legal forms?

**Review by BrandonPacewic — COMMENTED** [review]
Looks good and extremely clean, I left a few questions.

File: engine/Assets/Scripts/Analytics/AnalyticsManager.cs (hunk 1 above)
Comment (LucaHaverty) [reply engine/Assets/Scripts/Analytics/AnalyticsManager.cs:12]: Nope, I will remove the comment

File: engine/Assets/Scripts/Analytics/AnalyticsManager.cs (hunk 3 above)
Comment (LucaHaverty) [reply engine/Assets/Scripts/Analytics/AnalyticsManager.cs:48]: Yeah I think so. If we need to, we can add a popup at the beginning to ask new users.

File: engine/Assets/Scripts/Analytics/AnalyticsManager.cs (hunk 2 above)
Comment (BrandonPacewic) [reply engine/Assets/Scripts/Analytics/AnalyticsManager.cs:1]: 👍

File: engine/Assets/Scripts/Analytics/AnalyticsManager.cs (hunk 3 above)
Comment (BrandonPacewic) [reply engine/Assets/Scripts/Analytics/AnalyticsManager.cs:46]: 👍

**Review by KyroVibe — APPROVED** [review]
Events reported, looks good. @PepperLola Could you test to see if you get the invalid events thing as well?

**Review by PepperLola — APPROVED** [review]
Everything looks like it works. I built the project and ran it with analytics disabled and enabled and only the events from when it was enabled went through. I didn't get any invalid events either.

**Pull request merged**